### ISSUE TITLE
proposal to add three new concepts related to seabed stress (version 2)

### ISSUE THREAD

**Issue by bricheno** [cf-convention/vocabularies#244]
**Proposer Lucy Bricheno (NOC MSM)** 
follow-on from issue #241
sea_floor_horizontal_stress
sea_floor_horizontal_stress_due_to_sea_surface_waves
sea_floor_horizontal_stress_due_to_model_sea_water_velocity

- **Term** sea_floor_horizontal_stress
- **Description** Ocean processes such as waves and currents can generate forces (or stress) on the sea floor. The surface called "sea floor" means the lower boundary of the ocean. "Stress" means the shear stress (force per unit area) exerted at the sea floor.
- **Units** N m-2.

- **Term** sea_floor_horizontal_stress_due_to_sea_surface_waves
- **Description** Ocean processes such as waves and currents can generate forces (or stress) on the sea floor. The surface called "sea floor" means the lower boundary of the ocean. "Stress" means the shear stress (force per unit area) exerted at the sea floor. The specification of a physical process by the phrase "due_to_" process means that the quantity named is a single term in a sum of terms which together compose the general quantity named by omitting the phrase. The phrase "sea_surface_waves" means the stress associated with sea surface waves dissipation processes through bottom friction.
- **Units** N m-2.

- **Term** sea_floor_horizontal_stress_due_to_model_sea_water_velocity
- **Description** Ocean processes such as waves and currents can generate forces (or stress) on the sea floor. The surface called "sea floor" means the lower boundary of the ocean. "Stress" means the shear stress (force per unit area) exerted at the sea floor. The specification of a physical process by the phrase "due_to_" process means that the quantity named is a single term in a sum of terms which together compose the general quantity named by omitting the phrase. The phrase "model_sea_water_velocity" means the stress associated with ocean current dissipation processes through bottom friction.
- **Units** N m-2.

**Comment by JonathanGregory** [comment]
Thanks, Lucy. This all looks good to me. For the last term, I would suggest adding some more text, explaining why `model` is in the standard name: "This standard name is intended for the output of models in which sea surface waves are not included in the velocity field. Hence the sea floor stress due to bottom friction excludes any contribution from the dissipation of sea surface waves. If that contribution to the sea floor stress is separately modelled, it can be given the standard name of `sea_floor_horizontal_stress_due_to_sea_surface_waves`." Would that be OK?

**Comment by bricheno** [comment]
yes - that sounds perfect - thanks for adding this clarification.

**Comment by efisher008** [comment]
Dear Lucy @bricheno,

Thank you for your proposal and especially thanks for your patience, I know this has taken a while to be added to the editor so apologies about that. The entries have now been added and can be viewed here:

- `sea_floor_horizontal_stress` - https://cfeditor.ceda.ac.uk/proposal/5622
- `sea_floor_horizontal_stress_due_to_sea_surface_waves` - https://cfeditor.ceda.ac.uk/proposal/5623
- `sea_floor_horizontal_stress_due_to_model_sea_water_velocity` - https://cfeditor.ceda.ac.uk/proposal/5624

I have added Jonathan's clarification to the last name. I would also suggest adding the phrase "Horizontal stress refers to the stress in the horizontal plane" to each name. Is this alright? 

I have noticed the only existing standard name referring to horizontal stress, `sea_ice_average_normal_horizontal_stress`, has units of `N m-1`. Could you please explain the dimensionality of the stress component in these variable names - is this per unit area, or per unit length along the sea floor?

We could also add the following phrases from the definition to the CF phrasebank for use in later names:
- `sea_floor`: The surface called "sea floor" means the lower boundary of the ocean.
- `seafloor && stress`: "Stress" means the shear stress (force per unit area) exerted at the sea floor.

Best regards,
Ellie

**Comment by bricheno** [comment]
Hi Ellie, thanks for your work on this

Yes, I see that sea ice stress is in N/m and our new seabed stresses are in N/m2. Horizontal stress normally has units of Nm-2, so I wonder if, for the sea-ice, they are representing a vertically integrated stress. In sea ice modeling, sea ice is often treated as a thin 2D layer (because it's only a few meters thick)? could that explain things.. Maybe I need to look more at the sea-ice definition?

adding the new terms to the phrasebank sounds like a good suggestion too, hopefully these are not too contentious, and will be useful in future?

Best wishes,
Lucy


________________________________

**Comment by efisher008** [comment]
Hi Lucy,

Thank you for your explanation. The definition for the existing name `sea_ice_average_normal_horizontal_stress` is the following:
_""Sea ice" means all ice floating in the sea which has formed from freezing sea water, rather than by other processes such as calving of land ice to form icebergs. **Axial stress** is the symmetric component of the tensor representing the gradient of internal forces (e.g. in ice). Horizontal stress refers to the stress in the horizontal plane. "Horizontal" refers to the local horizontal in the location of the sea ice, i.e., perpendicular to the local gravity vector. **Average normal stress** refers to the average of the diagonal elements of the stress tensor and represents the first invariant of stress."_
This would imply that it is the force per unit area in the horizonal direction which is represented by the quantity, as it is a normal stress term. Therefore the dimensionality and canonical units should be distinct from your names, and so everything looks fine (according to my understanding!).

For reference, `sea_ice_average_normal_horizontal_stress` was proposed via mailing list in 2018 for the CMIP6 data request, and relates to the variable short name _sistresave_. 

I've added the definition of "horizontal" and "horizontal stress" to each name, and think these are now ready for finalising. I've also added the phrases `sea_floor`, `sea_floor_stress` and `sea_surface_waves` with their definitions to the CF phrasebank, which will ensure consistent usage of these phrases in future names. 

Do you have any further comments before I add the label "accept within 7 days" to this issue, and start the countdown to acceptance?

Best wishes,
Ellie

**Comment by bricheno** [comment]
Thanks for explaining Ellie,

I am happy with the definitions and you can 'press go' for the submission.
Lucy


________________________________

**Comment by efisher008** [comment]
Hi Lucy @bricheno,

7 days have passed and these three names have been accepted. They will be published in v91 of the Standard Names table, which is now planned for early May. 

Best wishes,
Ellie

**Comment by bricheno** [comment]
Thanks Ellie,

I've started populating my published datasets with these names. The idea next is to check that the datasets pass compliance checks, using cfchecks (cf-checker in python). Currently, I get these warnings:

But should these go away, once the v91 table has been published?
Also, one more thing - do you know if I also need to add header information regarding to "Conventions"?
My full output from the chchecks below FYI

Many thanks,
Lucy




CHECKING NetCDF FILE: Jan2090_hourly_combinedstress.nc
=====================
Using CF Checker Version 4.1.0
Checking against CF Version CF-1.8
Using Standard Name Table Version 90 (2025-03-20T01:16:14Z)
Using Area Type Table Version 13 (20 March 2025)
Using Standardized Region Name Table Version 5 (12 November 2024)

WARN: (2.6.1): No 'Conventions' attribute present

------------------
Checking variable: longitude
------------------

------------------
Checking variable: latitude
------------------

------------------
Checking variable: Mean_sea_floor_horizontal_stress
------------------
ERROR: (3.3): Invalid standard_name: sea_floor_horizontal_stress
INFO: attribute title is being used in a non-standard way

------------------
Checking variable: Maximum_sea_floor_horizontal_stress
------------------
ERROR: (3.3): Invalid standard_name: sea_floor_horizontal_stress
INFO: attribute title is being used in a non-standard way

------------------
Checking variable: sea_floor_horizontal_stress_due_to_sea_surface_waves
------------------
ERROR: (3.3): Invalid standard_name: sea_floor_horizontal_stress
INFO: attribute title is being used in a non-standard way

------------------
Checking variable: sea_floor_horizontal_stress_due_to_model_sea_water_velocity
------------------
ERROR: (3.3): Invalid standard_name: sea_floor_horizontal_stress
INFO: attribute title is being used in a non-standard way

ERRORS detected: 4
WARNINGS given: 1
INFORMATION messages: 4


________________________________

**Comment by efisher008** [comment]
Hi Lucy @bricheno,

The `invalid_standard_name` error should go away once the names are published. I believe that the message "_Attribute title is being used in a non-standard way_" is related to this error and should also disappear, but @japamment or @feggleton may know otherwise.

The warning "_No Conventions attribute is present_" means that you likely have not specified which version of the CF Conventions your dataset conforms to. The current version of the CF Metadata Conventions is CF-1.12 (link [here](https://cfconventions.org/Data/cf-conventions/cf-conventions-1.12/cf-conventions.html)), and I would particularly recommend reading the sections _2. NetCDF Files and Components_ and _3. Description of the Data_ to check that variables and attributes are being treated appropriately in your datasets. Once satisfied that the dataset conforms to these Conventions, a global attribute `:Conventions` should be added, with the string value "CF-1.12" (or whichever is the most recent version).

I hope this is helpful, and please let me know if you have further questions?

Best wishes,
Ellie
